diff --git a/src/dbus.c b/src/dbus.c
index 741a1c5..c67d5fa 100644
--- a/src/dbus.c
+++ b/src/dbus.c
@@ -592,7 +592,7 @@ int dbus_init(void)
 
         owner_id = g_bus_own_name(G_BUS_TYPE_SESSION,
                                   FDN_NAME,
-                                  G_BUS_NAME_OWNER_FLAGS_NONE,
+                                  G_BUS_NAME_OWNER_FLAGS_REPLACE,
                                   dbus_cb_bus_acquired,
                                   dbus_cb_name_acquired,
                                   dbus_cb_name_lost,
